src/utils/axiosInterceptors.ts

```diff
@@ -82,38 +80,54 @@ const fetchCachedResponse = async (request: any): Promise<any> => {
   return request;
 };
 
-const setCacheResponse = async (response: any): Promise<any> => {
-  if (response.config && response.config.useCache) {
-    let cache;
-    try {
-      cache = await caches.open(SENTINEL_HUB_CACHE);
-    } catch (err) {
-      console.warn('Caching failed', err);
-      return response;
-    }
-
-    let cacheKey;
-
-    const expiresMs = new Date().getTime() + EXPIRES_IN_SECONDS * 1000;
-    response.headers = {
-      ...response.headers,
-      [EXPIRY_HEADER_KEY]: expiresMs,
-    };
+const saveCacheResponse = async (response: any): Promise<any> => {
+  // not using cache?
+  if (!response.config.useCache) {
+    return response;
+  }
+  // do not perform caching if Cache API is not supported:
+  if (typeof window === 'undefined' || !window.caches) {
+    return response;
+  }
+  // resource not cacheable?
+  const cacheKey = await generateCacheKey(response.config);
+  if (cacheKey === null) {
+    return response;
+  }
+  // Cache API not supported?
```

Comment:
I think this doesn't belong here; on line 88, there's also a similar comment `  // do not perform caching if Cache API is not supported:`

Reply:
:+1:, fixed.

Reply:
Resolving.